src/components/content/monitor/Monitor.tsx

```diff
@@ -0,0 +1,282 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import 'echarts/lib/chart/bar';
+import '../../../styles/monitor.css';
+import { MonitorOutlined } from '@ant-design/icons';
+import { Button, Col, Form, Input, Row, Select, Tabs } from 'antd';
+import { Tab } from 'rc-tabs/lib/interface';
+import { useEffect, useState } from 'react';
+import MonitorChart from './MonitorChart';
+import { Metric, MonitorService, ServiceService, ServiceVo } from '../../../xpanse-api/generated';
+import { arrayQueue, Queue } from '../../utils/Queue';
+
+export interface MetricProps {
+    id: string;
+
+    name: string;
+
+    value: number;
+
+    vmName: string;
+}
+
+function Monitor(): JSX.Element {
+    const [form] = Form.useForm();
+    const [serviceId, setServiceId] = useState<string>('');
+    const [monitorChartItems, setMonitorChartItems] = useState<Tab[]>([]);
+    const [data, setData] = useState<ServiceVo[]>([]);
+    const [timer, setTimer] = useState<NodeJS.Timeout>();
+    const queue: Queue<Metric[]> = arrayQueue<Metric[]>(12);
```

Comment:
we can move the queue size 12 to constants file.

Reply:
fixed.